compiler/resolution/functionResolution.cpp

```diff
@@ -469,15 +471,19 @@ resolveUninsertedCall(BlockStmt* insideBlock,
   INT_ASSERT(block->parentSymbol);
 
   block->insertAtHead(call);
-  resolveCall(call);
+  if (checkonly && !call->primitive) {
+    resolveNormalCall(call, checkonly);
+  } else {
+    resolveCall(call);
```

Comment:
I don't see any reason why we couldn't have checkonly apply to primitives, it just seemed like an unnecessary amount of work to thread checkonly through all the other functions called by resolveCall, so we didn't do it.

Is your point that a caller of resolveUninsertedCall might miss that checkonly was being ignored in the case of primitives?  I think that is a fair concern, and so would be open to an INT_ASSERT with a comment about it being extensible or an INT_FATAL with that message.  @mppf - I suspect I don't have permission to push to your branch, how do you want to handle this?